src/prop/sat_proof_manager.h

```diff
@@ -582,6 +587,22 @@ class SatProofManager
   Node getClauseNode(const Minisat::Clause& clause);
   /** Prints clause, as a sequence of literals, in the "sat-proof" trace. */
   void printClause(const Minisat::Clause& clause);
+
+  /** The user context */
+  context::UserContext* d_userContext;
+
+  /** User-context dependent map from resolution conclusions to their assertion
+      level. */
+  context::CDHashMap<Node, int> d_optResLevels;
+  /** User-context-dependent map assertion level to proof nodes.
+   *
+   * This map is used to update the internal proof of this manager when the
+   * context pops.
+   */
+  std::map<int, std::vector<std::shared_ptr<ProofNode>>> d_optResProofs;
```

Comment:
This is not context-dependent?

Reply:
No, the documentation was wrong. This map is used by the optimized clauses manager, which is reinserting stuff in its parent proof from this map, according to how the user levels are popped.